lib/rules/require-has-block-helper.js

```diff
@@ -14,14 +16,31 @@ function getErrorMessage(name) {
 module.exports = class RequireHasBlockHelper extends Rule {
   visitor() {
     return {
-      PathExpression(node) {
-        if (TRANSFORMATIONS[node.original]) {
-          this.log({
-            message: getErrorMessage(node.original),
-            line: node.loc && node.loc.start.line,
-            column: node.loc && node.loc.start.column,
-            source: this.sourceForNode(node),
-          });
+      PathExpression(node, path) {
+        if (this.mode === 'fix') {
+          if (TRANSFORMATIONS[node.original]) {
+            let parent = path.parent;
+            if (
+              parent &&
+              (parent.node.type === 'BlockStatement' ||
+                (['MustacheStatement', 'SubExpression'].includes(parent.node.type) &&
+                  parent.node.path.original !== node.original))
+            ) {
```

Comment:
Let's break this conditional down into a few variables, to help simplify the mental model.

```suggestion
           let isBlockStatement = parent.node.type === 'BlockStatement';
           let isImplicitInvocation = ['MustacheStatement', 'SubExpression'].includes(parent.node.type) && parent.node.path.original !== node.original);
                  
            if (isBlockStatement || isImplicitInvocation) {
```

Reply:
Thanks, reads much better 👍